services/motion/motion.go

```diff
@@ -16,12 +20,83 @@ import (
 func init() {
 	resource.RegisterAPI(API, resource.APIRegistration[Service]{
 		RPCServiceServerConstructor: NewRPCServiceServer,
-		RPCServiceHandler:           servicepb.RegisterMotionServiceHandlerFromEndpoint,
-		RPCServiceDesc:              &servicepb.MotionService_ServiceDesc,
+		RPCServiceHandler:           pb.RegisterMotionServiceHandlerFromEndpoint,
+		RPCServiceDesc:              &pb.MotionService_ServiceDesc,
 		RPCClient:                   NewClientFromConn,
 	})
 }
 
+// GetPlanRequest describes the request to the GetPlan interface method.
+// Contains the ComponentName the returned plan(s) should be associated with,
+// an optional  ExecutionID and an Extra parameter.
+type GetPlanRequest struct {
+	ComponentName resource.Name
+	LastPlanOnly  bool
+	ExecutionID   uuid.UUID
+	Extra         map[string]interface{}
+}
+
+// PlanStep represents a single step of the plan
+// Describes the pose each resource described by the plan
+// should move to at that step.
+type PlanStep map[resource.Name]spatialmath.Pose
+
+// Plan represnts a motion plan.
+// Has a unique ID, ComponentName, ExecutionID and a sequence of Steps
+// which can be executed to follow the plan.
+type Plan struct {
+	ID            uuid.UUID
+	ComponentName resource.Name
+	ExecutionID   uuid.UUID
+	Steps         []PlanStep
+}
+
+// PlanState denotes the state a Plan is in.
+type PlanState uint8
```

Comment:
We may want other states in the future (waiting, planning, etc), prefer this just be a `int` or `uint` rather than starting it off with 5/8 slots already full.

Reply:
8 bit integers can represent 256 different values. I figured that it's unlikely we will have that many states.

Reply:
Wow. This is why I shouldn't review PRs the day after getting a covid shot, apparently.